sparsesim/src/lib.rs

```diff
@@ -178,34 +178,31 @@ impl QuantumSim {
     ///
     /// The function will panic if the given id does not correpsond to an allocated qubit.
     pub fn release(&mut self, id: usize) {
-        self.flush_queue(&[id], FlushLevel::HRxRy);
+        self.maybe_flush_queue(&[id], FlushLevel::HRxRy);
 
         let loc = self.id_map[id];
-        self.id_map.remove(id);
 
-        if self.id_map.is_empty() {
+        if self.id_map.iter().count() == 1 {
```

Comment:
Can you explain why you moved the removal of id to the end of this routine. It became a bit harder to understand this if.